experimental/tzif/src/parse/tzif.rs

```diff
@@ -0,0 +1,1540 @@
+                StandardWallIndicator::Standard,
+                StandardWallIndicator::Standard,
+                StandardWallIndicator::Wall,
+            ]
+        );
+    }
+
+    #[test]
+    fn parse_ut_local_indicators() {
+        // empty
+        assert_parse_err!(ut_local_indicators(3), "");
+
+        // invalid count
+        assert_parse_err!(ut_local_indicators(3), "\x00\x01");
+
+        // invalid standard-wall indicator
+        assert_parse_err!(ut_local_indicators(3), "\x00\x01\x02");
+
+        // valid standard-wall indicators
+        assert_parse_eq!(ut_local_indicators(0), "", Vec::<UtLocalIndicator>::new());
+        assert_parse_eq!(
+            ut_local_indicators(4),
+            "\x01\x00\x00\x01",
+            vec![
+                UtLocalIndicator::Ut,
+                UtLocalIndicator::Local,
+                UtLocalIndicator::Local,
+                UtLocalIndicator::Ut,
+            ]
+        );
+    }
+
+    #[test]
+    fn parse_footer() {
+        // missing leading newline
+        assert_parse_err!(footer(), "EST+5EDT,M3.2.0/2,M11.1.0/2\n");
+
+        // missing final newline
+        assert_parse_err!(footer(), "\nEST+5EDT,M3.2.0/2,M11.1.0/2");
+
+        // valid footer
+        assert_parse_eq!(
+            footer(),
+            "\nEST+5EDT,M3.2.0/2,M11.1.0/2\n",
+            PosixTzString {
+                std_info: ZoneVariantInfo {
+                    name: String::from("EST"),
+                    offset: Hours(5).as_seconds(),
+                },
+                dst_info: Some(DstTransitionInfo {
+                    variant_info: ZoneVariantInfo {
+                        name: String::from("EDT"),
+                        offset: Hours(4).as_seconds()
+                    },
+                    start_date: TransitionDate {
+                        day: TransitionDay::Mwd(3, 2, 0),
+                        time: Hours(2).as_seconds(),
+                    },
+                    end_date: TransitionDate {
+                        day: TransitionDay::Mwd(11, 1, 0),
+                        time: Hours(2).as_seconds(),
+                    },
+                })
+            }
+        );
+    }
+
+    fn parse_tzif_file<P: AsRef<Path>>(path: P) -> Result<(), ()> {
+        println!("parsing {:?}", path.as_ref().to_str());
+        let read = File::open(path).unwrap();
+        let stream = stream::buffered::Stream::new(
+            stream::position::Stream::new(stream::read::Stream::new(read)),
+            1,
+        );
+        let parsed = tzif().parse(stream);
+        assert!(parsed.is_ok());
+        //println!("{:#?}", parsed.unwrap().0);
+        Ok(())
+    }
+
+    #[test]
+    fn parse_tzif_testdata() -> Result<(), ()> {
+        for entry in WalkDir::new("testdata").follow_links(true) {
+            let entry = entry.unwrap();
+            if entry.file_type().is_file() {
+                parse_tzif_file(entry.path())?
+            }
+        }
+        Ok(())
+    }
+}
```

Comment:
This is technically an integration test, but I decided to just put it here so that I could make `walkdir` a `dev_dependency`.

But if anyone has a strong preference for actually moving this to a `tests` folder and then making `walkdir` an optional non-dev dependency, I'm happy to do that.